### DISCUSSION TITLE
How to overwrite default methods like invoke,stream_log etc.

### DISCUSSION THREAD

**Discussion by raksrivastava** [langchain-ai/langserve#495] [Q&A] [answered]
The rag-opensearch package is currently set to utilize OpenAI by default. However, I intend to switch it to Bedrock LLM and integrate AWS OpenSearch instead. **Could you guide me on how to override the default methods such as invoke and stream_log to achieve this**?

**Comment by eyurtsev** [comment, marked as answer]
Chat models all share the same interface in LangChain -- I am not sure why overwriting the default methods is necessary
